public/js/app.js

```diff
@@ -34,7 +48,7 @@ const App = {
       if ( !value )
         return;
 
-      request( "/todo", "POST", { text: value } )
+      request( "/todo", "POST", { text: value, listId: this.list.id } )
```

Comment:
Ich würde den Parameter `listId` als erstes, vor dem `text` der To-do setzen. Nach REST ist die Route ja wie folgt eigentlich, wenn du "Nested Resources" hast:
`/api/lists/:listId/todo`

Du muss es nicht verschachtelt machen, wie du es auch gemacht hast, auch wenn es hier durchaus sinnvoller wäre. Ist jedoch kein Muss.

Weiter unten betrifft es die Routen ebenfalls.